src/main/java/TransactionDao.java

```diff
@@ -0,0 +1,79 @@
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+
+class TransactionDao {
+
+    public void addTransaction(Transaction transaction) {
+        Connection connection = connect();
+
+        PreparedStatement preparedStatement = null;
+        try {
+            String sql = "INSERT INTO transactions(type, description, amount, dateStamp) VALUES (?, ?, ?, ?)";
+            preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setString(1, transaction.getTransactionType());
+            preparedStatement.setString(2, transaction.getDescription());
+            preparedStatement.setBigDecimal(3, transaction.getAmount());
+            preparedStatement.setDate(4, transaction.getDateStamp());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            System.out.println("Brak zapisu do bazy: " + e.getMessage());
+        }
+
+        closeConnection(connection);
+    }
+
+    public void modifyTransaction(Transaction transaction) {
+        Connection connection = connect();
+
+        PreparedStatement preparedStatement = null;
+        try {
+            String sql = "UPDATE transactions SET transactionType = ?, description = ?, amount = ?, dateStamp = ?) WHERE id = ?";
+            preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setString(1, transaction.getTransactionType());
+            preparedStatement.setString(2, transaction.getDescription());
+            preparedStatement.setBigDecimal(3, transaction.getAmount());
+            preparedStatement.setDate(4, transaction.getDateStamp());
+            preparedStatement.setInt(5, transaction.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            System.out.println("Brak zapisu do bazy: " + e.getMessage());
+        }
+
+        closeConnection(connection);
+    }
+
+    public List<Transaction> outcomes() {
+        return;
+    }//wspominales zeby robic to listami, ale dalej brakuje mi konceptu
```

Comment:
Nie różni to się mocno od selecta dla jednej wartości - tylko, że resultSet będzie miał nie 1 wartość zwracaną, tylko więcej. Dlatego musisz zrobić while(resultSet.next) - przmapuj na javową klasą i dodaj do listy. A potem zwracasz tą listę